libs/core/src/main/java/org/elasticsearch/core/internal/net/NetUtils.java

```diff
@@ -59,4 +64,45 @@
             return null;
         }
     }
+
+    /**
+     * If SO_KEEPALIVE is enabled (default), this method ensures sane default values for the extended socket options
+     * TCP_KEEPIDLE and TCP_KEEPINTERVAL. The default value for TCP_KEEPIDLE is system dependent, but is typically 2 hours.
+     * Such a high value can result in firewalls eagerly closing these connections. To tell any intermediate devices that
+     * the connection remains alive, we explicitly set these options to 5 minutes if the defaults are higher than that.
+     */
+    public static void setSaneDefaultKeepAliveOptions(NetworkChannel socketChannel) {
```

Comment:
Naming nit, suggest this instead:

```suggestion
    public static void tryEnsureReasonableKeepAliveConfig(NetworkChannel socketChannel) {
```